### PR TITLE
Document ViewTransitions API reference

### LINKED ISSUE #7961: Document View Transition events fields
### 📚 Subject area/topic

View Transitions

### 📋 Suggested page

Either on the View Transitions page (https://docs.astro.build/en/guides/view-transitions/#lifecycle-events) or on some reference page

### 📋 General description or bullet points (if proposing new content)

Document the type of the View Transition events, which fields are available and what each of them represent.

`astro:page-load`, `astro:after-preparation` and `astro:after-swap` have no special field and are the most used events, this may lead users to incorrectly assume that none of the events have special properties.

Both `astro:before-preparation` and `astro:before-swap` have [the following properties](https://github.com/withastro/astro/blob/237e17c8feea244b06b00ceaed08cb16d3d6baeb/packages/astro/src/transitions/events.ts#L21-L27):
- `from: URL`: The URL of the page initiating the transition.
- `to: URL`: The URL transitioning to. (can be modified)
- `direction: Direction | string`: Direction of the transition, used to change to a different animation when going back in history.
- `navigationType: NavigationTypeString`: What kind of history navigation is being applied
	- `push`: New URL is added to the history
	- `replace`: New URL replaces the previous URL in the history
	- `traverse`: History remains unchanged, transition is moving through the history
- `sourceElement: Element | undefined`: Element that triggered the navigation, if it came from a DOM element. This is undefined when the navigations was initiated from the `navigate` API.
- `info: any`: ?? Unclear to me where this comes from, requires some investigation.
- `newDocument: Document`: Content that will be swapped in place of the current document (`document.documentElement`).

`astro:before-preparation` also has:
- `formData: FormData | undefined`: The data that will be sent on the request to load the next page if a form action triggered the navigation.
- `loader: () => Promise<void>`: The function that runs the next phase of the transition (loading the next page). Can be overridden to add extra behavior or to prevent the execution of the next phase.

`astro:before-swap` also has:
- `viewTransition: ViewTransition`: The View Transition element that allows you to hook into it's promises. On browsers that don't have native VT this will be an object implementing the same API.
- `swap: () => void`: The function that runs the next phase of the transition (swapping the new document in place of the old document). Can be overridden to add extra behavior or to prevent the execution of the next phase.

### PR DESCRIPTION
**Pull request by Fryuni**
#### Description

Add a reference to the APIs and events available when using View Transitions.

#### Related issues & labels (optional)

- Closes #7961

### CODE REVIEW

**Comment by sarah11918** [comment]
Thank you @ArmandPhilippot for that great consistency work! So appreciated! 💪 

I think it's time for @martrapp to have a look at what we've all been up to.... 👀

File: src/content/docs/en/reference/api-reference.mdx [hunk 1]
```diff
@@ -2075,6 +2075,357 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+**Type:** [`ViewTransition`](https://developer.mozilla.org/en-US/docs/Web/API/ViewTransition)
+</p>
+
+The view transition object used in this navigation. On browser that do not support the [View Transitions API](https://developer.mozilla.org/en-US/docs/Web/API/View_Transitions_API), this is an object implementing the same API for convenience but without the DOM integration.
+
+#### `swap`
+
+<p>
+  *Type:* `() => void`
```
Comment (ArmandPhilippot) [inline src/content/docs/en/reference/api-reference.mdx:2372]: Same as above:
* the new line avoids a possible bug due to markdown used in HTML tags
* Type should be bold, not italic
```suggestion

**Type:** `() => void`
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 2]
```diff
@@ -2075,6 +2075,357 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+**Type:** `Document`
+</p>
+
+The document for the next page in the navigation. The contents of this document will be swapped in place of the contents of the current document.
+
+#### `navigationType`
+
+<p>
+  *Type:* `'push' | 'replace' | 'traverse'`
```
Comment (ArmandPhilippot) [inline src/content/docs/en/reference/api-reference.mdx:2295]: Same as the last review:
* the new line avoids a possible bug due to markdown used in HTML tags
* Type should be bold, not italic
```suggestion

**Type:** `'push' | 'replace' | 'traverse'`
```

**Review by ArmandPhilippot — COMMENTED** [review]
It seems I missed two incorrectly formatted `Type:` in the previous review. 👀

**Comment by sarah11918** [comment]
Noting that the link should work after the new View Transitions docs have merged tomorrow!

File: src/content/docs/en/reference/api-reference.mdx [hunk 3]
```diff
@@ -2075,6 +2075,359 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+When submiting an HTML form with View Transitions enabled, this field is automatically set to the data in the form. When using the [`navigate()` function](#navigate), this value is the same as given in the options.
+
+#### `loader`
+
+<p>
+
+**Type:** `() => Promise<void>`
+</p>
+
+Implementation of the following phase in the navigation (loading the next page). This implementation can be overriden to add extra behavior or to prevent the execution of the next phase.
```
Comment (martrapp) [inline src/content/docs/en/reference/api-reference.mdx:2355]: > ... or to prevent the execution of the next phase

That is a mystery to me. Do we have any examples of this anywhere? Why would you do that? And why does it cancel the next phase? Calling `preventDefault()` for the `astro:before-preparation` event, on the other hand, is fine.

If we do not persuade users of this, we do not need to warn them in the following

File: src/content/docs/en/reference/api-reference.mdx [hunk 4]
```diff
@@ -2075,6 +2075,359 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+Marks scripts in the new document that should not be executed. Those scripts are already in the current document and are not flagged for re-execution using [`data-astro-rerun`](/en/guides/view-transitions/#data-astro-rerun).
+
+##### `swapRootAttributes()`
+
+<p>
+
+**Type:** `(newDocument: Document) => void`
+</p>
+
+Swaps the Astro-injected internal attributes between the document roots, like the document scope hash.
```
Comment (martrapp) [inline src/content/docs/en/reference/api-reference.mdx:2399]:
```suggestion
Swaps the attributes between the document roots, like the `lang` attribute. This also includes Astro-injected internal attributes like `data-astro-transition`, which makes the transition direction available to Astro-generated CSS rules.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 5]
```diff
@@ -2075,6 +2075,359 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+When making a custom swap function, it is important to call this function. Such attributes are used for Client Islands hydration and CSS scoping.
```
Comment (martrapp) [inline src/content/docs/en/reference/api-reference.mdx:2401]:
```suggestion
  When making a custom swap function, it is important to call this function so as not to break the view transition animations.
```

**Review by martrapp — COMMENTED** [review]
Wow! My wholehearted thanks to the Arcane Tower 🧙🏼‍♂️ and everyone who helped. I've just added a few small suggestions that you may or may not want to consider.

**Comment by sarah11918** [comment]
Just resolved conflicts/updated branch! With any luck, that means the link check now passes!

**Comment by sarah11918** [comment]
Needed to resolve conflicts again because this page just updated! :joy:

File: src/content/docs/en/reference/api-reference.mdx [hunk 6]
```diff
@@ -2137,6 +2137,359 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+<p>
+
+**Type:** `() => Promise<void>`
+</p>
+
+Implementation of the following phase in the navigation (loading the next page). This implementation can be overriden to add extra behavior or to prevent the execution of the next phase.
+
+:::caution
+Overriding this implementation to prevent the execution may leave the browser on an inconsistent navigation state. It is normally preferred to use `preventDefault()` if you just want to cancel the navigation and stay in the current page.
+:::
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:2421]:
```suggestion
Implementation of the following phase in the navigation (loading the next page). This implementation can be overriden to add extra behavior.
```
You never have to convince me to remove a caution! Executive decision suggestion!

File: src/content/docs/en/reference/api-reference.mdx [hunk 7]
```diff
@@ -2137,6 +2137,359 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+**Type:** `'auto' | 'push' | 'replace'`<br />
+**Default:** `'auto'`<br />
+<Since v="3.2.0" />
+</p>
+
+Defines how this navigation should be added to the browser history.
+
+- `'push'`: the router will use `history.pushState` to create a new entry in the browser history.
+- `'replace'`: the router will use `history.replaceState` to update the URL without adding a new entry into navigation.
+- `'auto'` (default): the router will attempt `history.pushState`, but if the URL is not one that can be transitioned to, the current URL will remain with no changes to the browser history.
```
Comment (yanthomasdev) [inline src/content/docs/en/reference/api-reference.mdx:2173]: "is not one that" is a bit wordy, I believe we can just change to "cannot"?
```suggestion
- `'auto'` (default): the router will attempt `history.pushState`, but if the URL cannot be transitioned to, the current URL will remain with no changes to the browser history.
```

**Review by sarah11918 — APPROVED** [review]
Thank you so much @Fryuni for taking the initiative here! Would not be possible without the outstanding contributions of @ArmandPhilippot and the wise oversight of @martrapp .

Best Docs Team Ever! :partying_face:

File: src/content/docs/en/reference/api-reference.mdx [hunk 8]
```diff
@@ -2137,6 +2137,359 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+This option follows the [`history` option](https://developer.mozilla.org/en-US/docs/Web/API/Navigation/navigate#history) from the browser Navigation API, but simplified for the cases that can happen on an Astro project.
```
Comment (yanthomasdev) [inline src/content/docs/en/reference/api-reference.mdx:2175]: comma nit
```suggestion
This option follows the [`history` option](https://developer.mozilla.org/en-US/docs/Web/API/Navigation/navigate#history) from the browser Navigation API but simplified for the cases that can happen on an Astro project.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 9]
```diff
@@ -2137,6 +2137,355 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+**Type:** `FormData`<br />
+<Since v="3.5.0" />
+</p>
+
+A `FormData` object for `POST` requests.
+
+When this option is given, the requests to the navigation target page will be sent as a `POST` request with the given form data object as the content.
+
+Submiting an HTML form with View Transitions enable will use this method in place of the default navigation with page reload. Calling this method allows triggering the same behavior programatically.
```
Comment (yanthomasdev) [inline src/content/docs/en/reference/api-reference.mdx:2189]: typos
```suggestion
Submitting an HTML form with View Transitions enabled will use this method in place of the default navigation with page reload. Calling this method allows triggering the same behavior programmatically.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 10]
```diff
@@ -2137,6 +2137,355 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+#### `info` option
+
+<p>
+
+**Type:** `any`<br />
+<Since v="3.6.0" />
+</p>
+
+Arbitrary data to be included in the `astro:before-preparation` and `astro:before-swap` event caused by this navigation.
```
Comment (yanthomasdev) [inline src/content/docs/en/reference/api-reference.mdx:2199]:
```suggestion
Arbitrary data to be included in the `astro:before-preparation` and `astro:before-swap` events caused by this navigation.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 11]
```diff
@@ -2137,6 +2137,355 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+<p>
+
+**Type:** `() => 'none' | 'animate' | 'swap'`<br />
+<Since v="3.6.0" />
+</p>
+
+Returns the fallback strategy for View Transitions.
+
+See the guide on [Fallback control](/en/guides/view-transitions/#fallback-control) for how to chose and configure the fallback behavior.
```
Comment (yanthomasdev) [inline src/content/docs/en/reference/api-reference.mdx:2257]:
```suggestion
See the guide on [Fallback control](/en/guides/view-transitions/#fallback-control) for how to choose and configure the fallback behavior.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 12]
```diff
@@ -2137,6 +2137,355 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+An event dispatched after the next page in a navigation using View Transitions is loaded.
+
+This event has no attributes.
+
+Read more about how to use this event on the [View Transitions guide](/en/guides/view-transitions/#astroafter-preparation).
+
+### `astro:before-swap`
+
+An event dispatched after the next page is parsed and prepared and linked into a document in preparation for the transition but before any content is swapped between the documents.
```
Comment (yanthomasdev) [inline src/content/docs/en/reference/api-reference.mdx:2290]: wording nit
```suggestion
An event dispatched after the next page is parsed, prepared, and linked into a document in preparation for the transition but before any content is swapped between the documents.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 13]
```diff
@@ -2137,6 +2137,355 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+An event dispatched after the contents of the page have been swapped but before the view transition ends.
+
+The history entry and scroll position have already been updated when this event is triggered.
+
+### `astro:page-load`
+
+An event dispatched after a page completes loading, be it from a navigation using View Transition or native to the browser.
+
+When view transitions is enabled in the page, code that would normally execute on `DOMContentLoaded` should be changed to execute on this event.
```
Comment (yanthomasdev) [inline src/content/docs/en/reference/api-reference.mdx:2317]: We're being a bit inconsistent by using both "view transitions" and "View Transitions", the latter appearing more often. I'd make all occurrences the same for consistency's sake. Also, a few nits.

```suggestion
An event dispatched after the contents of the page have been swapped but before the view transition ends.

The history entry and scroll position have already been updated when this event is triggered.

### `astro:page-load`

An event dispatched after a page completes loading, be it from a navigation using View Transitions or native to the browser.

When View Transitions is enabled on the page, code that would normally execute on `DOMContentLoaded` should be changed to execute on this event.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 14]
```diff
@@ -2137,6 +2137,355 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+#### `formData`
+
+<p>
+
+**Type:** `FormData | undefined`
+</p>
+
+A `FormData` object for `POST` requests.
+
+When this attribute is set, a `POST` requests will be sent to the [`to` URL](#to) with the given form data object as the content instead of the normal `GET` request.
```
Comment (yanthomasdev) [inline src/content/docs/en/reference/api-reference.mdx:2406]:
```suggestion
When this attribute is set, a `POST` request will be sent to the [`to` URL](#to) with the given form data object as the content instead of the normal `GET` request.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 15]
```diff
@@ -2137,6 +2137,355 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+When submiting an HTML form with View Transitions enabled, this field is automatically set to the data in the form. When using the [`navigate()` function](#navigate), this value is the same as given in the options.
```
Comment (yanthomasdev) [inline src/content/docs/en/reference/api-reference.mdx:2408]:
```suggestion
When submitting an HTML form with View Transitions enabled, this field is automatically set to the data in the form. When using the [`navigate()` function](#navigate), this value is the same as given in the options.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 16]
```diff
@@ -2137,6 +2137,355 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+#### `loader`
+
+<p>
+
+**Type:** `() => Promise<void>`
+</p>
+
+Implementation of the following phase in the navigation (loading the next page). This implementation can be overriden to add extra behavior.
```
Comment (yanthomasdev) [inline src/content/docs/en/reference/api-reference.mdx:2417]:
```suggestion
Implementation of the following phase in the navigation (loading the next page). This implementation can be overridden to add extra behavior.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 17]
```diff
@@ -2137,6 +2137,355 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+#### `viewTransition`
+
+<p>
+
+**Type:** [`ViewTransition`](https://developer.mozilla.org/en-US/docs/Web/API/ViewTransition)
+</p>
+
+The view transition object used in this navigation. On browser that do not support the [View Transitions API](https://developer.mozilla.org/en-US/docs/Web/API/View_Transitions_API), this is an object implementing the same API for convenience but without the DOM integration.
```
Comment (yanthomasdev) [inline src/content/docs/en/reference/api-reference.mdx:2426]:
```suggestion
The View Transition object used in this navigation. On browsers that do not support the [View Transitions API](https://developer.mozilla.org/en-US/docs/Web/API/View_Transitions_API), this is an object implementing the same API for convenience but without the DOM integration.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 18]
```diff
@@ -2137,6 +2137,355 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+<p>
+
+**Type:** `() => void`
+</p>
+
+Implementation of the document swap logic. 
+
+Read more on how to [build your own custom swap function](/en/guides/view-transitions/#building-a-custom-swap-function) on the View Transitions guide.
+
+By default this implementation will call the following functions in order:
```
Comment (yanthomasdev) [inline src/content/docs/en/reference/api-reference.mdx:2439]:
```suggestion
By default, this implementation will call the following functions in order:
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 19]
```diff
@@ -2137,6 +2137,355 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+##### `swapRootAttributes()`
+
+<p>
+
+**Type:** `(newDocument: Document) => void`
+</p>
+
+Swaps the attributes between the document roots, like the `lang` attribute. This also includes Astro-injected internal attributes like `data-astro-transition`, which makes the transition direction available to Astro-generated CSS rules.
+
+When making a custom swap function, it is important to call this function so as not to break the view transition animations.
```
Comment (yanthomasdev) [inline src/content/docs/en/reference/api-reference.mdx:2459]:
```suggestion
When making a custom swap function, it is important to call this function so as not to break the View Transition's animations.
```

**Review by yanthomasdev — COMMENTED** [review]
Good work! I have mostly nit and typo suggestions here, after applying them I believe it's good to go!

File: src/content/docs/en/reference/api-reference.mdx (hunk 13 above)
Comment (sarah11918) [reply src/content/docs/en/reference/api-reference.mdx]: view transitions always lower case UNLESS it's the proper name of the View Transitions API. (which we are probably renaming in 5.0 anyway...) I'll go through and do a sweep for this!

File: src/content/docs/en/reference/api-reference.mdx [hunk 20]
```diff
@@ -2137,6 +2137,355 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+<Since v="3.6.0" />
+</p>
+
+Returns the fallback strategy for View Transitions.
+
+See the guide on [Fallback control](/en/guides/view-transitions/#fallback-control) for how to choose and configure the fallback behavior.
+
+## View Transition Events
+
+<p><Since v="3.6.0" /></p>
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:2261]:
```suggestion
```
Executive decisions b/c of TOC. This can't be an `h2`, and if it's an `h3` then we don't see these events in the sidebar. 

May revisit in future (when we rename the API for 5.0 anyway!) to show `h4` in the TOC on reference pages.

File: src/content/docs/en/reference/api-reference.mdx [hunk 21]
```diff
@@ -2137,6 +2137,355 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+- [`navigationType`](#navigationtype)
+- [`direction`](#direction)
+- [`from`](#from)
+- [`to`](#to)
+- [`formData`](#formdata)
+- [`loader`](#loader)
+
+Read more about how to use this event on the [View Transitions guide](/en/guides/view-transitions/#astrobefore-preparation).
+
+### `astro:after-preparation`
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:2280]:
```suggestion
### `astro:after-preparation` event
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 22]
```diff
@@ -2137,6 +2137,355 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+An event dispatched after the next page in a navigation using View Transitions is loaded.
+
+This event has no attributes.
+
+Read more about how to use this event on the [View Transitions guide](/en/guides/view-transitions/#astroafter-preparation).
+
+### `astro:before-swap`
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:2288]:
```suggestion
### `astro:before-swap` event
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 23]
```diff
@@ -2137,6 +2137,355 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+- [`navigationType`](#navigationtype)
+- [`direction`](#direction)
+- [`from`](#from)
+- [`to`](#to)
+- [`viewTransition`](#viewtransition)
+- [`swap`](#swap)
+
+Read more about how to use this event on the [View Transitions guide](/en/guides/view-transitions/#astrobefore-swap).
+
+### `astro:after-swap`
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:2307]:
```suggestion
### `astro:after-swap` event
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 24]
```diff
@@ -2137,6 +2137,355 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+An event dispatched after the contents of the page have been swapped but before the view transition ends.
+
+The history entry and scroll position have already been updated when this event is triggered.
+
+### `astro:page-load`
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:2313]:
```suggestion
### `astro:page-load` event
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 25]
```diff
@@ -2137,6 +2137,355 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+### `astro:before-swap`
+
+An event dispatched after the next page is parsed, prepared, and linked into a document in preparation for the transition but before any content is swapped between the documents.
+
+This event can't be canceled. Calling `preventDefault()` is a no-op.
+
+This event has the attributes:
+- [`info`](#info)
+- [`state`](#state-option)
+- [`sourceElement`](#sourceelement-option)
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:2297]:
```suggestion
- [`state`](#state)
- [`sourceElement`](#sourceelement)
```

**Review by sarah11918 — COMMENTED** [review]
some quick tidying

File: src/content/docs/en/reference/api-reference.mdx [hunk 26]
```diff
@@ -2137,6 +2137,352 @@ export const onRequest = defineMiddleware(async (context, next) => {
 })
 ```
 
+## View Transitions client API (`astro:transitions/client`)
+
+<p><Since v="3.2.0" /></p>
+
+This module provides functions to control and interact with the View Transitions and client-side router.
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:2144]:
```suggestion
This module provides functions to control and interact with the View Transitions API and client-side router.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 27]
```diff
@@ -2137,6 +2137,352 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+### `navigate()`
+
+<p>
+
+**Type:** `(href: string, options?: Options) => void`<br />
+<Since v="3.2.0" />
+</p>
+
+Start a navigation to the given `href` using View Transitions.
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:2156]:
```suggestion
A function that starts a navigation to the given `href` using the View Transitions API.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 28]
```diff
@@ -2137,6 +2137,352 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+<p>
+
+**Type:** `FormData`<br />
+<Since v="3.5.0" />
+</p>
+
+A `FormData` object for `POST` requests.
+
+When this option is given, the requests to the navigation target page will be sent as a `POST` request with the given form data object as the content.
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:2187]:
```suggestion
When this option is provided, the requests to the navigation target page will be sent as a `POST` request with the form data object as the content.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 29]
```diff
@@ -2137,6 +2137,352 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+Submitting an HTML form with View Transitions enabled will use this method instead of the default navigation with page reload. Calling this method allows triggering the same behavior programmatically.
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:2189]:
```suggestion
Submitting an HTML form with view transitions enabled will use this method instead of the default navigation with page reload. Calling this method allows triggering the same behavior programmatically.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 30]
```diff
@@ -2137,6 +2137,352 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+### `supportsViewTransitions`
+
+<p>
+
+**Type:** `boolean`<br />
+<Since v="3.2.0" />
+</p>
+
+A constant that indicates whether View Transitions are supported and enabled in the current browser.
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:2235]:
```suggestion
Whether or not view transitions are supported and enabled in the current browser.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 31]
```diff
@@ -2137,6 +2137,352 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+### `transitionEnabledOnThisPage`
+
+<p>
+
+**Type:** `boolean`<br />
+<Since v="3.2.0" />
+</p>
+
+Whether the current page has the View Transitions enabled for client-side navigation. This can be used to make reusable components that adapt to when they are used on pages with and without View Transitions.
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:2245]:
```suggestion
Whether or not the current page has view transitions enabled for client-side navigation. This can be used to make components that behave differently when they are used on pages with view transitions.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 32]
```diff
@@ -2137,6 +2137,352 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+### `getFallback`
+
+<p>
+
+**Type:** `() => 'none' | 'animate' | 'swap'`<br />
+<Since v="3.6.0" />
+</p>
+
+Returns the fallback strategy for View Transitions.
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:2255]:
```suggestion
Returns the fallback strategy to use in browsers that do not support view transitions.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 33]
```diff
@@ -2137,6 +2137,352 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+
+### `astro:before-preparation` event
+
+An event dispatched at the beginning of a navigation using View Transitions. This event happens before any request is made and any browser state is changed.
+
+This event has the attributes:
+- [`info`](#info)
+- [`state`](#state-option)
+- [`sourceElement`](#sourceelement-option)
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:2267]:
```suggestion
- [`state`](#state)
- [`sourceElement`](#sourceelement)
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 34]
```diff
@@ -2137,6 +2137,352 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+
+### `astro:after-swap` event
+
+An event dispatched after the contents of the page have been swapped but before the view transition ends.
+
+The history entry and scroll position have already been updated when this event is triggered.
+
+### `astro:page-load` event
+
+An event dispatched after a page completes loading, be it from a navigation using view transitions or native to the browser.
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:2312]:
```suggestion
An event dispatched after a page completes loading, whether from a navigation using view transitions or native to the browser.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 35]
```diff
@@ -2137,6 +2137,352 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+<p>
+
+**Type:** `FormData | undefined`
+</p>
+
+A `FormData` object for `POST` requests.
+
+When this attribute is set, a `POST` request will be sent to the [`to` URL](#to) with the given form data object as the content instead of the normal `GET` request.
+
+When submitting an HTML form with View Transitions enabled, this field is automatically set to the data in the form. When using the [`navigate()` function](#navigate), this value is the same as given in the options.
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:2405]:
```suggestion
When submitting an HTML form with view transitions enabled, this field is automatically set to the data in the form. When using the [`navigate()` function](#navigate), this value is the same as given in the options.
```
